LiteCore/Query/QueryParser.cc

```diff
@@ -352,11 +352,50 @@ namespace litecore {
         return count;
     }
 
+namespace {
+    bool canDefaultTableHaveDeleted() { return true; }
+
+    bool needPatchDeleteFlag(const string& table, QueryParser::DeletionStatus delStatus) {
+        auto pos = table.rfind('_');
+        if (pos == string::npos || table.substr(pos + 1) != "default") {
+            return false;
+        }
+        // We only need to patch the delete flag for kLiveDocs, because the other two
+        // DeletionStatus are deduced from the query itself, and they must explicitly
+        // appear in the query expression.
+        return canDefaultTableHaveDeleted() && delStatus == QueryParser::kLiveDocs;
+    }
+} // anonymous namespace
 
     void QueryParser::writeWhereClause(const Value *where) {
-        if (where) {
+        auto& aliasInfo = _aliases[_dbAlias];
+        bool patchDeleteFlag = needPatchDeleteFlag(aliasInfo.tableName, aliasInfo.delStatus);
+
+        if (!where && !patchDeleteFlag) {
+            // We don't have where and don't need patch the delete flag.
+            return;
+        }
+
+        _checkedDeleted = false;
+        if (patchDeleteFlag) {
             _sql << " WHERE ";
+        }
+        if (where) {
```

Comment:
I think this can be merged with the if above like `if (where || patchDeleteFlag)`?

Reply:
not straightforward. I wanted it to turn back to the current code if patchDeleteFlag is false.

Reply:
OK